.azure-pipelines/sync-fork.yml

```diff
@@ -0,0 +1,54 @@
+parameters:
+  - name: fork_repo_url
+    displayName: fork_repo_url
+    default: https://github.com/microsoft/openjdk-jdk
+  - name: upstream_repo_url
+    displayName: upstream_repo_url
+    default: https://github.com/openjdk/jdk
+  - name: fork_branch
+    default: main
+    displayName: fork_branch
+  - name: upstream_branch
+    default: master 
+    displayName: upstream_branch
+
+trigger: none
+
+schedules:
+  # At 5am UTC (10pm PST) daily
+  - cron: "0 5 * * *"
+    displayName: Sync
+    always: false # only run if there has been code changes to the jdk codebase
+    branches:
+      include:
+        - "*"
+stages:
+- stage: sync
+  jobs:
+  - job: sync
+    displayName: "sync repository with upstream"
+
+    pool:
+      name: JEG-ubuntu20.04-x64-EO
+    
+    steps:    
+      - bash: |
+          fork_repo_url=${{ parameters.fork_repo_url }}
+          repourl="${fork_repo_url#https://github.com/schandra13/}"
+          echo "reponame is : $repourl"
+          echo "##vso[task.setvariable variable=FORK_FOLDER]$repourl"
+        displayName: "Fetch fork repo folder name"
+
+      - bash: |
+          git clone ${{ parameters.fork_repo_url }}
```

Comment:
Each line in this task can be separated into its own task. That is a _general_ rule of thumb for Azure DevOps pipelines. The tasks fail/succeed based on the very last EXITCODE, and it's very useful for our team to know _precisely_ where our pipeline fails to quicken our ability to determine root cause/mitigation.

Reply:
separated tasks as required